app/src/main/java/com/example/instagram/MainActivity.java

```diff
@@ -35,178 +39,37 @@
 public class MainActivity extends AppCompatActivity {
 
     public static final String TAG = "MainActivity";
-    public final static int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
-    private Button btnLogout;
-    private EditText etDescription;
-    private Button btnCaptureImage;
-    private ImageView ivPostImage;
-    private Button btnSubmit;
-    public String photoFileName = "photo.jpg";
-    private File photoFile;
+
+    final FragmentManager fragmentManager = getSupportFragmentManager();
```

Comment:
I looked up the `FragmentActivity` source and this should be OK, but my knee-jerk reaction was that this call makes me nervous as it is called when the class is instantiated but not necessarily after Activity `onCreate`. If this was Kotlin I would suggest using [lazy initialization](https://kotlinlang.org/docs/delegated-properties.html). Like I said this looks to be fine just keep it in mind in the future.